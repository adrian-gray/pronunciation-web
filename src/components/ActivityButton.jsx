import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";

import Chip from "@material-ui/core/Chip";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

import { capitalise } from "./../utils/utils";

const useStyles = makeStyles(theme => ({
  chip: {
    margin: theme.spacing(1)
  },
  open: {
    color: "green"
  },
  member: {
    color: "#BBB"
  },
  pending: {
    color: "maroon"
  },
  page: theme.page
}));

const toLink = (phoneme, activity) => {
  const title = capitalise(activity);
  const websafe = activity.split(" ").join("-");
  const link = `/sound/${phoneme}/${websafe}`;
  return { title, link };
};

const ActivityButton = props => {
  const classes = useStyles(props);

  const { accessStatus, activity, phoneme, subscriptionLevel } = props;
  const { title, link } = toLink(phoneme, activity);

  let chipClass = classes[accessStatus];

  const [chipClasses, setChipClasses] = useState(
    `${classes.chip} ${chipClass}`
  );

  useEffect(() => {
    if (accessStatus === "member" && subscriptionLevel) {
      chipClass = classes["open"];
      setChipClasses(`${classes.chip} ${chipClass}`);
    }
  });

  return (
    <Chip
      label={title}
      className={chipClasses}
      component={Link}
      to={link}
      clickable
      key={title}
    />
  );
};

export default withTheme(ActivityButton);
