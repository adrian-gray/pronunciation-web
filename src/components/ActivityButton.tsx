import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";

import { capitalise } from "./../utils/utils";

const toLink = (phoneme, activity) => {
  const title = capitalise(activity);
  const websafe = activity.split(" ").join("-");
  const link = `/sound/${phoneme}/${websafe}`;
  return { title, link };
};

export default props => {
  const { accessStatus, activity, phoneme, subscriptionLevel } = props;
  const { title, link } = toLink(phoneme, activity);

  const [chipClasses, setChipClasses] = useState(
    `chip activity-${accessStatus}`
  );

  useEffect(() => {
    if (accessStatus === "member") {
      if (subscriptionLevel) {
        setChipClasses(`chip activity-open`);
      } else {
        setChipClasses(`chip activity-locked`);
      }
    }
  });

  return (
    <Link title={title} className={chipClasses} to={link} key={title}>
      {title}
    </Link>
  );
};
