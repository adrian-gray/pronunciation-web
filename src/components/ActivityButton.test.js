import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import ActivityButton from "./ActivityButton";

afterEach(cleanup);

test("ActivityButton", async () => {
  const props = {
    accessStatus: undefined,
    subscriptionLevel: 0,
    activity: "harry",
    phoneme: "a:"
  };

  const container = shallow(<ActivityButton {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
