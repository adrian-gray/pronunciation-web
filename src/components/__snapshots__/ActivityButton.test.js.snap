// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityButton 1`] = `undefined`;
